File: pinyougou-parent/pinyougou-pay-interface/src/main/java/com/pinyougou/pay/service/PayService.java
package com.pinyougou.pay.service;

import com.pinyougou.pojo.TbPayLog;

import java.util.Map;

public interface PayService {
    /**
     * 支付链接
     * @return
     */
    Map<String, String> createNative(String out_trade_no,String total_fee);
    /**
     * 查询支付状态
     * @param out_trade_no
     */
    public Map queryPayStatus(String out_trade_no);

    /**
     * 根据用户查询 payLog
     * @param userId
     * @return
     */
    public TbPayLog searchPayLogFromRedis(String userId);

    /**
     * 关闭支付
     * @param out_trade_no
     * @return
     */
    public Map closePay(String out_trade_no);
}

File: pinyougou-parent/pinyougou-seckill-web/src/main/webapp/js/service/payService.js
app.service("payService",function ($http) {
    //请求后端生成二维码url
    this.createPayUri = function () {
        return $http.get("pay/createPayUri.do");
    }
    this.queryPayStatus=function(out_trade_no){
        return $http.get('pay/queryPayStatus.do?out_trade_no='+out_trade_no);
    }
})